app/javascript/orangelight/figgy_manifest_manager.js

```diff
@@ -0,0 +1,207 @@
+
+import loadResourcesByBibId from './load-resources-by-bib-id'
+
+class FiggyViewer {
+  constructor(idx, element, manifestUrl) {
+    this.idx = idx
+    this.element = element
+    this.manifestUrl = manifestUrl
+  }
+
+  getAvailabilityElement() {
+    let elements = document.querySelectorAll('#availability > div.location--panel.location--online > div > div.panel-body > div')
+    if (elements.length < 1) {
+      elements = document.querySelectorAll('#availability > div.location--panel.location--online > div > div.panel-body > div > ul > div.electronic-access')
+    }
+    // This assumes that the first element is the link
+    const element = elements[0]
+    return element
+  }
+
+  getArkLinkElement() {
+    // If there is only one electronic access link, it's structure using a <div> rather than <ul>
+    const availabilityElement = this.getAvailabilityElement()
+
+    if (!availabilityElement) {
+      return null
+    }
+
+    let elements = availabilityElement.querySelectorAll('div > a')
+    if (elements.length < 1) {
+      elements = availabilityElement.querySelectorAll('li > a')
+    }
+    return elements[this.idx]
+  }
+
+  updateArkLinkElement() {
+    const arkLinkElement = this.getArkLinkElement()
+    if (!arkLinkElement) {
+      return
+    }
+
+    arkLinkElement.href = `#viewer-${this.idx + 1}`
```

Comment:
We have many, many ARKs registered with `#view` URLs — can we preserve that id for the first viewer?